### PR TITLE
Hook up react-query fetch cancellation

### PR DESCRIPTION
**Pull request by Zeko369**
Closes: #3555

## Feature Checklist

- [x] Pass `signal` from `useQuery` to our `__internal_buildRpcClient` to be able to cancel queries on unmount

### CODE REVIEW

File: packages/blitz-rpc/src/data-client/rpc.ts [hunk 1]
```diff
@@ -93,9 +93,6 @@ export function __internal_buildRpcClient({
       serialized = serialize(params)
     }
 
-    // Create a new AbortController instance for this request
-    const controller = new AbortController()
-
```
Comment (beerose) [inline packages/blitz-rpc/src/data-client/rpc.ts:98]: Hmm, shouldn't we still use this in case `signal` argument is undefined?

Comment (Zeko369) [reply packages/blitz-rpc/src/data-client/rpc.ts:98]: Unless we want to wire it up to something, this won't do anything by itself so 🤷‍♂️

**Review by beerose — APPROVED** [review]

**Comment by beerose** [comment]
I forgot to mention that before merging, but we also need docs for that. We ideally shouldn't change the PR template

**Comment by beerose** [comment]
Hey @Zeko369, are you going to add docs for that, or you want me to do it?

**Pull request merged**